src/Handler/CurlFactory.php

```diff
@@ -495,12 +495,14 @@ private static function retryFailedRewind(
 
     private function createHeaderFn(EasyHandle $easy)
     {
-        if (!isset($easy->options['on_headers'])) {
```

Comment:
This was only rewritten so phpstorm understands that `$onHeaders` can be a callable or null and does not mark this with an error.